### PR TITLE
Update ref handling and adjust label component indentation

### PR DESCRIPTION
**Pull request by Hanxbeen**
- Updated the ref handling condition from `Object(ref) === ref` to `'current' in ref` for improved clarity and accuracy. This change simplifies the check for object refs and aligns with common React practices.
- Adjusted the indentation of the label component for consistency and readability. This helps maintain a clean and uniform code style throughout the component.

Closes #<issue_number>

#### Changelog

**New**

- N/A

**Changed**

- Ref handling condition updated from `Object(ref) === ref` to `'current' in ref` for improved clarity and alignment with React practices.
- Indentation of the `<label>` component adjusted for consistency and readability.

**Removed**

- N/A

#### Testing / Reviewing

To verify these changes:
1. Review the updated ref handling logic in the `Checkbox` component to ensure it properly distinguishes between function and object refs.
2. Verify that the modifications to the `<label>` start tag are correctly applied, ensuring there are no syntax errors or unexpected issues, and confirm that the tag displays properly.

### CODE REVIEW

**Comment by Hanxbeen** [comment]
> Thanks for your submission! We ask that you sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> I have read the DCO document and I hereby sign the DCO.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **DCO Assistant Lite bot**.

I have read the DCO document and I hereby sign the DCO.

**Comment by Hanxbeen** [comment]
> Thanks for your submission! We ask that you sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> I have read the DCO document and I hereby sign the DCO.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **DCO Assistant Lite bot**.

recheck

**Comment by Hanxbeen** [comment]
> Thanks for your submission! We ask that you sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> I have read the DCO document and I hereby sign the DCO.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **DCO Assistant Lite bot**.



> > Thanks for your submission! We ask that you sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> > I have read the DCO document and I hereby sign the DCO.
> > You can retrigger this bot by commenting **recheck** in this Pull Request. Posted by the **DCO Assistant Lite bot**.
> 
> recheck



> * Updated the ref handling condition from `Object(ref) === ref` to `'current' in ref` for improved clarity and accuracy. This change simplifies the check for object refs and aligns with common React practices.
> * Adjusted the indentation of the label component for consistency and readability. This helps maintain a clean and uniform code style throughout the component.
> 
> Closes #<issue_number>
> 
> #### Changelog
> **New**
> 
> * N/A
> 
> **Changed**
> 
> * Ref handling condition updated from `Object(ref) === ref` to `'current' in ref` for improved clarity and alignment with React practices.
> * Indentation of the `<label>` component adjusted for consistency and readability.
> 
> **Removed**
> 
> * N/A
> 
> #### Testing / Reviewing
> To verify these changes:
> 
> 1. Review the updated ref handling logic in the `Checkbox` component to ensure it properly distinguishes between function and object refs.
> 2. Verify that the modifications to the `<label>` start tag are correctly applied, ensuring there are no syntax errors or unexpected issues, and confirm that the tag displays properly.

I have read the DCO document and I hereby sign the DCO.

recheck

**Comment by Hanxbeen** [comment]
> * Updated the ref handling condition from `Object(ref) === ref` to `'current' in ref` for improved clarity and accuracy. This change simplifies the check for object refs and aligns with common React practices.
> * Adjusted the indentation of the label component for consistency and readability. This helps maintain a clean and uniform code style throughout the component.
> 
> Closes #<issue_number>
> 
> #### Changelog
> **New**
> 
> * N/A
> 
> **Changed**
> 
> * Ref handling condition updated from `Object(ref) === ref` to `'current' in ref` for improved clarity and alignment with React practices.
> * Indentation of the `<label>` component adjusted for consistency and readability.
> 
> **Removed**
> 
> * N/A
> 
> #### Testing / Reviewing
> To verify these changes:
> 
> 1. Review the updated ref handling logic in the `Checkbox` component to ensure it properly distinguishes between function and object refs.
> 2. Verify that the modifications to the `<label>` start tag are correctly applied, ensuring there are no syntax errors or unexpected issues, and confirm that the tag displays properly.

I have read the DCO document and I hereby sign the DCO.

recheck

**Comment by Hanxbeen** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by Hanxbeen** [comment]
recheck

**Review by preetibansalui — APPROVED** [review]

File: packages/react/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -200,7 +200,8 @@ const Checkbox = React.forwardRef(
         <label
           htmlFor={id}
           className={`${prefix}--checkbox-label`}
-          title={title}>
+          title={title}
+        >
```
Comment (emyarod) [inline packages/react/src/components/Checkbox/Checkbox.tsx]:
```suggestion
          title={title}>
```

I believe this is enforced by the prettier formatting rules in the repo

Comment (Hanxbeen) [reply packages/react/src/components/Checkbox/Checkbox.tsx]: then, should i rollback this line?

Comment (tay1orjones) [reply packages/react/src/components/Checkbox/Checkbox.tsx]: Running `yarn format` from the root will run prettier against the whole codebase if you don't have prettier installed in your editor. I just did this and updated the PR so this should be good to go now. Sorry for the confusion!

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**